src/widgets/ot_color_picker.eliom

```diff
@@ -0,0 +1,183 @@
+                         ["#FF7E02"; "#FFC503"];
+                         ["#01CD64"; "#AF58B9"];
+                         ["#0198DD"; "#254760"];
+                         ["#FFFFFF"; "#000000"]]]
+
+let color_samples_6 = [[["#BEC3C7"; "#7A8E8D"];
+                        ["#1C3D50"; "#0280B4"];
+                        ["#00A385"; "#A444B2"]]]
+
+
+(* Take one list of list of list of color (string) and build table list with it.
+   return also div_color_list to allow to launch start script detection. *)
+let generate_color_table color_samples =
+
+  let build_color_div color =
+    D.div ~a:[a_class["ot-color-picker-square"];
+            a_title color;
+            a_style ("background-color: " ^ color ^ ";")]
+      []
+  in
+  let build_td_color color_div =
+    td ~a:[a_class["ot-color-picker-table-td"]] [color_div]
+  in
+  let build_tr_color tds =
+    tr ~a:[a_class["ot-color-picker-table-tr"]] tds
+  in
+
+  let rec build_table div_color_list tables = function
+    | []                -> div_color_list, tables
+    | head::tail        ->
+
+      let rec build_column div_color_list trs = function
+        | []            -> div_color_list, trs
+        | head::tail    ->
+
+          let rec build_line div_color_list tds = function
+            | []                -> div_color_list, tds
+            | color::tail       ->
+
+              let color_div = build_color_div color in
+              build_line
+                (color_div::div_color_list)
+                ((build_td_color color_div)::tds)
+                tail
+          in
+
+          let div_color_list', tds = build_line div_color_list [] head in
+          build_column
+            div_color_list'
+            ((build_tr_color tds)::trs)
+            tail
+      in
+
+      let div_color_list', trs = build_column div_color_list [] head in
+      let tbl = table ~a:[a_class["ot-color-picker-table"]] trs in
+      build_table
+        div_color_list'
+        (tbl::tables)
+        tail
+  in
+
+  let div_color_list, tables = build_table [] [] color_samples in
+  div_color_list, tables
+
+let make ?(initial_color = 0, 0, 0) ?(color_samples = Lazy.force color_samples_p5) () =
+  let tbl, trl, tdl = initial_color in
+  let color_ref = ref (List.nth (List.nth (List.nth color_samples tbl) trl) tdl) in
+  let div_color_list, tables = generate_color_table color_samples in
+  let color_div = D.div ~a:[a_class["ot-color-picker-current-color"];
+                          a_title !color_ref;
+                          a_style ("background-color: " ^ !color_ref ^ ";")] []
+  in
+  let block = D.div ~a:[a_class["ot-color-picker-block"]] tables in
+  let type_t = (color_ref, color_div, div_color_list, block) in
+  type_t, color_div, block
+
+]
+
+[%%client
+
+open Lwt
+
+let fusion (color_ref, color_div, fst_list, block) (_, _, snd_list, _) =
+  (color_ref, color_div, fst_list@snd_list, block)
+
+let start (color_ref, color_div, color_list, _) =
```

Comment:
Can `start` happen automatically? I don't think we have similar `start` methods in the other Ocsigen Toolkit widgets.